certbot-ci/certbot_integration_tests/conftest.py

```diff
@@ -0,0 +1,115 @@
+                              'pebble-nonstrict', 'pebble-strict'],
+                     help='select the ACME server to use (boulder-v1, boulder-v2, '
+                          'pebble-nonstrict or pebble-strict), '
+                          'defaulting to pebble-nonstrict')
+
+
+def pytest_configure(config):
+    """
+    Standard pytest hook used to add a configuration logic for each node of a pytest run.
+    :param config: the current pytest configuration
+    """
+    if not hasattr(config, 'slaveinput'):
+        with _print_on_err():
+            _setup_integration_tests(config)
+
+    if not os.environ.get('CERTBOT_ACME_TYPE'):
+        raise ValueError('Error, CERTBOT_ACME_TYPE environment variable is not set !')
+    config.acme_xdist = _get_acme_xdist()
+
+
+def _get_acme_xdist():
+    """
+    Get the acme server config distribution from the environment variable "CERTBOT_ACME_XDIST"
+    :return: a dict of the acme server config distribution
+    """
+    acme_xdist = os.environ.get('CERTBOT_ACME_XDIST')
+    if not acme_xdist:
+        raise ValueError('Error, CERTBOT_ACME_XDIST environment variable is not set !')
+
+    return json.loads(acme_xdist)
+
+
+@contextlib.contextmanager
+def _print_on_err():
+    """
+    With pytest-xdist, stdout is used for nodes communication, so print is uneffective.
+    However, stderr is still available. This context manager transfers stdout to stderr
+    for the duration of the context, allowing to display prints to the user.
+    """
+    old_stdout = sys.stdout
+    sys.stdout = sys.stderr
+    try:
+        yield
+    finally:
+        sys.stdout = old_stdout
+
+
+def _setup_integration_tests(config):
+    """
+    Setup the environment for integration tests.
+    Will:
+        - check runtime compatiblity (Docker, docker-compose, Nginx)
+        - create a temporary workspace and the persistent GIT repositories space
+        - configure and start paralleled ACME CA servers using Docker
+        - transfer ACME CA servers configurations to pytest nodes using env variables
+    :param config: Configuration of the pytest master node
+    """
+    # Check for runtime compatibility: some tools are required to be available in PATH
+    try:
+        subprocess.check_output(['docker', '-v'], stderr=subprocess.STDOUT)
+    except (subprocess.CalledProcessError, OSError):
+        raise ValueError('Error: docker is required in PATH to launch the integration tests, '
+                         'but is not installed or not available for current user.')
+
+    try:
+        subprocess.check_output(['docker-compose', '-v'], stderr=subprocess.STDOUT)
+    except (subprocess.CalledProcessError, OSError):
+        raise ValueError('Error: docker-compose is required in PATH to launch the integration tests, '
+                         'but is not installed or not available for current user.')
+
+    workers = ['master'] if not config.option.numprocesses\
```

Comment:
Yes, it will work. None is a falsy value, so ['master'] will be taken in this case. This allow tests to run even if pytest-xdist is not present or enabled.

Reply:
I meant "this all" as in "the whole setup", rather than this particular line. What's the best way to test that? Manually set `numprocesses` to 0?

Reply:
Oh OK. It is the default situation in fact. If you remember, we did the change to avoid to set numprocesses to auto in certbot for a pytest execution. So if you do not set it, you are in the situation of only primary around, doing the tests. Or you set it explicitly, or execute the tox target that set it also.

I tested all cases if success of course.